From 18def77d27f88ce26b6af29fe56a80429fed555d Mon Sep 17 00:00:00 2001
From: Ville Voutilainen <ville.voutilainen@qt.io>
Date: Tue, 15 Nov 2022 03:28:11 +0200
Subject: Fix dangling references

These were found with the help of -Wdangling-reference, which is
new in GCC 13.

The one in qtpaths.cpp is a false positive: parseLocationOrError()
returns a reference, so there's nothing for the full expression to
destroy. Moreover, it returns a reference to a static object, so
there's no destruction inside the function either.

The other two aren't, but are also harmless. QDBusMessage::arguments()
and QVariant::toList() return a stored QVariantList by value, so
QList's COW mechanism means at() returns a reference that will not be
destroyed. However, the compiler has no way of knowing that. And
since it depends on the implementation details, change the code to
not depend on that.


Pick-to: 6.5
Change-Id: If53aa16fcc24586d752ffc76c193c81e43dc9d95
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
---
 src/tools/qtpaths/qtpaths.cpp | 5 +++++
 1 file changed, 5 insertions(+)

(limited to 'src/tools/qtpaths')

diff --git a/src/tools/qtpaths/qtpaths.cpp b/src/tools/qtpaths/qtpaths.cpp
index b4e2d749aa..fa29381d78 100644
--- a/src/tools/qtpaths/qtpaths.cpp
+++ b/src/tools/qtpaths/qtpaths.cpp
@@ -252,6 +252,10 @@ int main(int argc, char **argv)
         results << typesList.join('\n');
     }
 
+    QT_WARNING_PUSH
+#if defined(Q_CC_GNU_ONLY) && Q_CC_GNU >= 1300 && Q_CC_GNU < 1400
+    QT_WARNING_DISABLE_GCC("-Wdangling-reference")
+#endif
     if (parser.isSet(display)) {
         const StringEnum &location = parseLocationOrError(parser.value(display));
         QString text = QStandardPaths::displayName(location.enumvalue);
@@ -303,6 +307,7 @@ int main(int argc, char **argv)
         QStringList paths = QStandardPaths::locateAll(location.enumvalue, searchitem, QStandardPaths::LocateFile);
         results << location.mapName(paths.join(pathsep));
     }
+    QT_WARNING_POP
 
 #if !QT_CONFIG(settings)
     if (parser.isSet(query) || parser.isSet(qtconf) || parser.isSet(queryformat)) {
-- 
cgit v1.2.1

